### ISSUE TITLE
DNN: Exception "inputs.size() >= 2" on calling Forward

### ISSUE THREAD

**Issue by steffenxonna** [opencv/opencv#16319]
### System information (version)
- OpenCV => 4.2.0 (via OpenCVSharp https://github.com/shimat/opencvsharp realesed on 08.01.2020)
- Operating System / Platform => Windows 64 Bit
- Compiler => Visual Studio 2017 (C#)

### Detailed description
I have tried to use FlowNet 2.0 (https://github.com/lmb-freiburg/flownet2) to calculate the Optical Flow between two images. I'm setting two inputs and changed the Resolutions in the prototxt.template to the resolution i want to use. Every time i call Forward i got the Exception "inputs.size() >= 2". 

### Steps to reproduce
* Downloaded the model "flownet2-models.tar.gz" contained in the shell script https://github.com/lmb-freiburg/flownet2/blob/master/models/download-models.sh
* Using the largest model contained in the folder "FlowNet2".
* Changed the Placeholder Variables (Resolutions) in the prototext to the one i want to use. Based on the logic in https://github.com/lmb-freiburg/flownet2/blob/master/scripts/run-flownet.py
  * My variant of the file: [FlowNet2_deploy.prototxt.txt](https://github.com/opencv/opencv/files/4045088/FlowNet2_deploy.prototxt.txt)
* Run Forward like the C#-Code below.

left and right are some images read from an video-file or just two images loaded with ImRead. I think i should not attach some files during copyright problems.

    using (var net = CvDnn.ReadNetFromCaffe(@"FlowNet2_deploy.prototxt.txt",
        @"FlowNet2_weights.caffemodel.h5"))
    {
        net.SetPreferableBackend(Net.Backend.OPENCV);
        net.SetPreferableTarget(Net.Target.CPU);

        var img0 = CvDnn.BlobFromImage(left, 1.0, new Size(1024, 720), new Scalar(), true, false);
        var img1 = CvDnn.BlobFromImage(right, 1.0, new Size(1024, 720), new Scalar(), true, false);
        net.SetInput(img0, "img0");
        net.SetInput(img1, "img1");
    
        var predict = net.Forward("predict_flow_final");  << exception
    }

I have used an face-detection model based on Caffe with one input and this worked fine.

**Comment by dkurt** [comment]
```yaml
layer {
  name: "Eltwise1"
  type: "Eltwise"
  bottom: "img0"
  top: "img0s"
  eltwise_param {
    operation: SUM
    coeff: 0.00392156862745098
  }
}
```

Looks like it's expected that output of this operation returns `0.0039 * img0`. So only multiplication is performed.

Please try to replace it to `Power` layer with corresponding value of `scale`.

**Comment by steffenxonna** [comment]
Okay. This seems to bring them a bit forward. But the he are unable to create an layer of type "DataAugmentation". This type is currently not supported, right?
Same for layer type "Resample"?

**Comment by dkurt** [comment]
I believe there is a workaround with removing DataAugmentation and Resample layerf from prototxt and doing it using `cv::resize` out of the network (not sure for now what `DataAugmentation` does). 

Anyway it must be great sample for OpenCV.

**Comment by steffenxonna** [comment]
Thanks. That's an good idea.

`Resample` could be done outside. The largest model uses an resample nearly at the end. So i switched to the smallest version of the model. Also to reduce the complexity. Getting things work first.

`DataAugmentation` should enhance the data set. They should apply random modifications to the input data to give more variations of the same image. The parameter `augment_during_test: true` sound like the augmentation will only applied during the test phase. I'm a bit confused but hopefully the layer should not be needed. I think.

After this changes i got the error `inputs.size() == (size_t)1`. I think this will be raised from the `Convolutional`-Layer. Because they are using two inputs and two outputs. I have splitted these layers into two seperate layer. But now i'm getting the Exception `blobs.size() != 0`.

But i think splitting up these layers also could breaks the weights and the functionallity of the model. One `Convolutional`-Layer with mutliple inputs should apply one set of filters to all inputs. An multiple `Convolutional`-Layers should create an set of filter for each Layer.

Latest Version of the prototxt i'm using.
[FlowNet2-C_deploy.prototxt.txt](https://github.com/opencv/opencv/files/4050495/FlowNet2-C_deploy.prototxt.txt)

**Comment by l-bat** [comment]
@steffenxonna We need caffe reference to test some FlowNet2 layers. Can you run [script](https://gist.github.com/l-bat/7cb4f10c174d20d75b8cb6c2ac407991) and provide `out_dir` with outputs? You need to download [proto.zip](https://github.com/opencv/opencv/files/4266078/proto.zip).
how to run script:
`python flownet2_save_data.py --proto_dir path/to/proto --output_dir path/to/out_dir`

**Comment by steffenxonna** [comment]
@l-bat I can try it. But i only used the pre-trained models with opencvsharp. I will need some days for it.

**Comment by steffenxonna** [comment]
@l-bat It tooks much more time than i previous expected. But after like 10 Linux installations and an new graphiccard i can serve you the data. I also included the console output in the zip-file.
[out.zip](https://github.com/opencv/opencv/files/4432610/out.zip)

Had to change something in your script. Flownet 2 uses an changed caffe-version. I found out that the new layers dosn't support cpu mode. They only work with cuda. Don't know if this could be an problem for opencv implementation?
Anyway i added the following lines. Before the usage of any net.
```
caffe.set_device(0)
caffe.set_mode_gpu()
```
